pkg/util/util.go

```diff
@@ -5,12 +5,85 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net/http"
+	"os"
+	"text/template"
+
+	f "github.com/flyteorg/flytectl/pkg/filesystemutils"
 )
 
 type githubversion struct {
 	TagName string `json:"tag_name"`
 }
 
+const (
+	AdminConfigTemplate = `admin:
+  # For GRPC endpoints you might want to use dns:///flyte.myexample.com
+  endpoint: {{.Host}}
+  authType: Pkce
+  insecure: {{.Insecure}}
+logger:
+  show-source: true
+  level: 0`
+	StorageConfigTemplate = `
+storage:
+  connection:
+    access-key: minio
+    auth-type: accesskey
+    disable-ssl: true
+    endpoint: http://localhost:30084
+    region: us-east-1
+    secret-key: miniostorage
+  type: minio
+  container: "my-s3-bucket"
+  enable-multicontainer: true`
+	StorageS3ConfigTemplate = `
+storage:
+  type: stow	
+  stow:
+    kind: s3
+    config:
+      auth_type: iam
+      region: <replace> # Example: us-east-2
+  container: <replace> # Example my-bucket. Flyte k8s cluster / service account for execution should have read access to this bucket`
+	StorageGCSConfigTemplate = `
+storage:
+  type: stow	
+  stow:
+    kind: s3
+    config:
+      json: ""
+      project_id: <replace-me> # TODO: replace <project-id> with the GCP project ID
+      scopes: https://www.googleapis.com/auth/devstorage.read_write
+  container: <replace> # Example my-bucket. Flyte k8s cluster / service account for execution should have read access to this bucket`
+)
+
+type ConfigTemplateValuesSpec struct {
+	Host     string
+	Insecure bool
+	Template string
```

Comment:
Template itself being part of the value spec seems odds,

Reply:
no I can change  the variable name and that would be ok